### PR TITLE
TOS-119: Badge

### PR DESCRIPTION
**Pull request by darktend**
## Ticket
Link to ticket:https://teamchallengeonlinestore.atlassian.net/jira/software/projects/TOS/boards/1?assignee=609eb5f09b362f0069a70820&selectedIssue=TOS-119

## Description

I created a badge, created a custom component for discounts, for new products and for marking whether the product is in stock

## Screenshots
image

(prefer animated gif)

## Checklist
- [x] Did you test it?
- [x] Did you add a screenshot?
- [x] Did you fix the merge conflict?
- [x] Did you fill the PR template?
- [x] Did you review this PR on your own?

#### Last step if the ABOVE STEPS are COMPLETED!
- [x] Did you share the link in a group chat?

### CODE REVIEW

File: app/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import cn from 'classnames';
+import React, { useMemo } from 'react';
+import styles from './Badge.module.scss';
+
+interface BadgesProps {
+    className?: string;
+    type?: 'default' | 'discount' | 'arrivals' | 'stock';
```
Comment (dev1B) [inline app/components/Badge/Badge.tsx:7]: Does "Out of stock" mean "stock"?

File: app/components/Badge/Badge.module.scss [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    &.discount {
+        background: #EC4F4F;
+    }
+
+    &.arrivals {
+        background: #EC914F;
+    }
+
+    &.stock {
+        max-width: 141px;
```
Comment (dev1B) [inline app/components/Badge/Badge.module.scss:52]: Do we need this line?

File: app/components/Badge/Badge.module.scss [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        font-size: 11px;
+        font-family: 'Roboto', sans-serif;
+        line-height: 16px;
+    }
+
+    &.discount,
+    &.arrivals,
+    &.stock {
+        width: 80px;
+        height: 22px;
```
Comment (dev1B) [inline app/components/Badge/Badge.module.scss:32]: Why did you add fixed sizes?

File: app/components/Badge/Badge.module.scss [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    top: 0;
+    right: 0;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    font-weight: 500;
+
+    &.default {
+        color: #393228;
+        border-radius: 100px;
```
Comment (dev1B) [inline app/components/Badge/Badge.module.scss:17]: Why did you add 100px?

**Review by dev1B — CHANGES_REQUESTED** [review]

File: app/components/Badge/Badge.module.scss [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    &.discount,
+    &.arrivals,
+    &.stock {
+        width: 80px;
+        height: 22px;
+        padding: 2px 12px;
+        flex-shrink: 0;
+        border-radius: 0px 10px;
+        color: #FEFEF9;
+        font-family: 'Montserrat', sans-serif;
```
Comment (dev1B) [inline app/components/Badge/Badge.module.scss:37]: Could you start using scss variables?

File: app/components/Badge/Badge.module.scss (hunk 2 above)
Comment (darktend) [reply app/components/Badge/Badge.module.scss:52]: add pading

File: app/components/Badge/Badge.tsx (hunk 1 above)
Comment (darktend) [reply app/components/Badge/Badge.tsx:7]: rename type

File: .eslintrc [hunk 6]
```diff
@@ -12,12 +12,12 @@
     "plugin:@next/next/recommended",
     "airbnb",
     "airbnb-typescript",
-    "plugin:@typescript-eslint/recommended-type-checked",
-    "plugin:@typescript-eslint/stylistic-type-checked",
+//    "plugin:@typescript-eslint/recommended-type-checked",
```
Comment (dev1B) [inline .eslintrc:15]: What do you plan to do with this?

File: app/components/Badge/Badge.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import cn from 'classnames';
+import React, { useMemo } from 'react';
+import styles from './Badge.module.scss';
+
+interface BadgesProps {
+  className?: string;
+  type?: 'essentials' | 'deals' | 'newbies' | 'inventory';
```
Comment (dev1B) [inline app/components/Badge/Badge.tsx:7]: @Taraskin777 @KarlAnna Do you like these types? Are these clear to you?

File: .eslintrc (hunk 6 above)
Comment (dev1B) [reply .eslintrc:15]: @Taraskin777 @KarlAnna Can we remove these?

File: app/components/Badge/Badge.tsx (hunk 7 above)
Comment (Taraskin777) [reply app/components/Badge/Badge.tsx:7]: as for me, not exactly

File: .eslintrc (hunk 6 above)
Comment (Taraskin777) [reply .eslintrc:15]: it's hard for me to say if this is necessary for our project because I haven't used these rules before

Comment (darktend) [reply .eslintrc:15]: @dev1B

**Review by dev1B — APPROVED** [review]

**Pull request merged**